LoopKitUI/Extensions/Environment+Dismiss.swift

```diff
@@ -0,0 +1,22 @@
+//
+//  Environment+Dismiss.swift
+//  Loop
+//
+//  Created by Michael Pangburn on 4/15/20.
+//  Copyright © 2020 LoopKit Authors. All rights reserved.
+//
+
+import SwiftUI
+
+
+private struct PresentationDismissalKey: EnvironmentKey {
+    static let defaultValue = {}
+}
+
+
+extension EnvironmentValues {
+    var dismiss: () -> Void {
+        get { self[PresentationDismissalKey.self] }
+        set { self[PresentationDismissalKey.self] = newValue }
+    }
+}
```

Comment:
interesting. So this allows dismissal of any `View`?

Reply:
Yeah. If a SwiftUI view is presented in a modal via UIKit code, the PresentationMode API doesn't work for dismissal; this is a lightweight way for a hosting controller to pass down a closure to dismiss.